test/e2e-migrated/metrics/cloud_provider_attributes_test.go

```diff
@@ -0,0 +1,96 @@
+		deploymentName = uniquePrefix("deployment")
+		genNs          = uniquePrefix("gen")
+		mockNs         = uniquePrefix("mock")
+	)
+
+	backend := kitbackend.New(mockNs, kitbackend.SignalTypeMetrics, kitbackend.WithName(backendName))
+	pipeline := testutils.NewMetricPipelineBuilder().
+		WithName(pipelineName).
+		WithRuntimeInput(true).
+		WithRuntimeInputContainerMetrics(true).
+		WithRuntimeInputPodMetrics(true).
+		WithRuntimeInputNodeMetrics(true).
+		WithRuntimeInputVolumeMetrics(true).
+		WithRuntimeInputDeploymentMetrics(false).
+		WithRuntimeInputStatefulSetMetrics(false).
+		WithRuntimeInputDaemonSetMetrics(false).
+		WithRuntimeInputJobMetrics(false).
+		WithOTLPOutput(testutils.OTLPEndpoint(backend.Endpoint())).
+		Build()
+	metricProducer := prommetricgen.New(genNs)
+
+	podSpec := telemetrygen.PodSpec(telemetrygen.SignalTypeMetrics)
+
+	deployment := kitk8s.NewDeployment(deploymentName, mockNs).WithPodSpec(podSpec).WithLabel("name", deploymentName).K8sObject()
+
+	resources := []client.Object{
+		kitk8s.NewNamespace(mockNs).K8sObject(),
+		kitk8s.NewNamespace(genNs).K8sObject(),
+		&pipeline,
+		metricProducer.Pod().WithPrometheusAnnotations(prommetricgen.SchemeHTTP).K8sObject(),
+		metricProducer.Service().WithPrometheusAnnotations(prommetricgen.SchemeHTTP).K8sObject(),
+		deployment,
+	}
+	resources = append(resources, backend.K8sObjects()...)
+
+	t.Cleanup(func() {
+		require.NoError(t, kitk8s.DeleteObjects(context.Background(), resources...)) //nolint:usetesting // Remove ctx from DeleteObjects
+	})
+	Expect(kitk8s.CreateObjects(t.Context(), resources...)).Should(Succeed())
+
+	assert.MetricPipelineHealthy(t.Context(), pipelineName)
+	assert.DeploymentReady(t.Context(), kitkyma.MetricGatewayName)
+	assert.DeploymentReady(t.Context(), backend.NamespacedName())
+	assert.DeploymentReady(t.Context(), types.NamespacedName{Name: deploymentName, Namespace: mockNs})
+	agentMetricsURL := suite.ProxyClient.ProxyURLForService(kitkyma.MetricAgentMetricsService.Namespace, kitkyma.MetricAgentMetricsService.Name, "metrics", ports.Metrics)
+	assert.EmitsOTelCollectorMetrics(t, agentMetricsURL)
+	backendContainsDesiredCloudResourceAttributes(t, backend, "cloud.region")
+	backendContainsDesiredCloudResourceAttributes(t, backend, "cloud.availability_zone")
+	backendContainsDesiredCloudResourceAttributes(t, backend, "host.type")
+	backendContainsDesiredCloudResourceAttributes(t, backend, "host.arch")
+	backendContainsDesiredCloudResourceAttributes(t, backend, "k8s.cluster.name")
+	backendContainsDesiredCloudResourceAttributes(t, backend, "cloud.provider")
+}
+
+func backendContainsDesiredCloudResourceAttributes(t *testing.T, backend *kitbackend.Backend, attribute string) {
+	t.Helper()
+
+	assert.BackendDataEventuallyMatches(t, backend,
+		HaveFlatMetrics(
+			ContainElement(SatisfyAll(
+				HaveResourceAttributes(HaveKey(attribute)),
+			)),
+		), fmt.Sprintf("could not find metrics matching resource attribute %s", attribute))
+}
```

Comment:
Can't we have a single BackendDataEventuallyMatches that checks if the attributes are present?